sdks/java/fn-execution/src/test/java/org/apache/beam/sdk/fn/data/BeamFnDataBufferingOutboundObserverTest.java

```diff
@@ -146,6 +159,85 @@ public void testConfiguredBufferLimit() throws Exception {
         Iterables.get(values, 1));
   }
 
+  @Test
+  public void testConfiguredTimeLimit() throws Exception {
+    Collection<BeamFnApi.Elements> values = new ArrayList<>();
+    AtomicBoolean onCompletedWasCalled = new AtomicBoolean();
+    PipelineOptions options = PipelineOptionsFactory.create();
+    options
+        .as(ExperimentalOptions.class)
+        .setExperiments(Arrays.asList("beam_fn_api_data_buffer_time_limit=1"));
+    CloseableFnDataReceiver<WindowedValue<byte[]>> consumer =
+        BeamFnDataBufferingOutboundObserver.forLocation(
+            options,
+            OUTPUT_LOCATION,
+            CODER,
+            TestStreams.withOnNext(addToValuesConsumer(values))
+                .withOnCompleted(setBooleanToTrue(onCompletedWasCalled))
+                .build());
+
+    // Test that it emits when time passed the time limit
+    consumer.accept(valueInGlobalWindow(new byte[1]));
+    Thread.sleep(10); // allow the flush thread to flush the buffer
```

Comment:
This test will be flaky since its based upon very small amounts of time. You want to make withOnNext free a lock/semaphore/countdownlatch and in your test you wait on it with a large time limit instead of using thread.sleep